src/rancilio-pid.cpp

```diff
@@ -1832,7 +1832,7 @@ void setup() {
         }
 
         if (MQTT == 1) {
-            snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "will");
+            snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "available");
```

Comment:
snprintf(topic_will, sizeof(topic_will), "%s%s/%s", mqtt_topic_prefix, hostname, "status");
I would change the topic name.
Compared to other mqtt stuff that i use, following would make more sense to me:
Topic will be "status", payload will be "online" or "offline"
Logic in my mind: If i want to get the availability i check the status of the mashine online or offline.
If the status of the mashine is online it is available.

Reply:
Sounds good to me. I changed the topic name of the will to status and tested it with my setup.

Reply:
In Home Assistant I have defined my temperature sensor as following:

```
  - name: Rancilio Silvia Temperature
    unique_id: sensor.silvia_temp
    availability:
      - payload_available: online
        payload_not_available: offline
        topic: espresso/rancilio.silvia/status
    state_topic: espresso/rancilio.silvia/temperature
    state_class: measurement
    device_class: temperature
    unit_of_measurement: °C
```

After switching my silvia off home assistant receives the will message and marks the sensor as unavailable.